backend/src/api/integration/helpers/devtoValidators.ts

```diff
@@ -0,0 +1,29 @@
+import PermissionChecker from '../../../services/user/permissionChecker'
+import Permissions from '../../../security/permissions'
+import ApiResponseHandler from '../../apiResponseHandler'
+import Error400 from '../../../errors/Error400'
+import { getUserByUsername } from '../../../serverless/integrations/usecases/devto/getUser'
+import { getOrganization } from '../../../serverless/integrations/usecases/devto/getOrganization'
+
+export default async (req, res) => {
+  try {
+    new PermissionChecker(req).validateHasAny([
+      Permissions.values.integrationCreate,
+      Permissions.values.integrationEdit,
+    ])
+
+    if (req.query.username) {
+      const result = await getUserByUsername(req.query.username)
+      await ApiResponseHandler.success(req, res, result)
+    } else if (req.query.organization) {
```

Comment:
This might have been on purpose (I haven't read the docs/specs), but I'll leave this comment just to double-check that we don't want to integrate/follow an organization if we're not integrating any user.

Reply:
So this is just to validate when creating an integration (onboarding or on settings page) to check whether the user entered valid Dev.to organizations or Dev.to users. 

Initially validating was called immediately after the user entered some text (denounced 300ms) but that turned out not to work as nice as I would hoped and I switched so that the validation is done when the user clicks `Connect`. 

This API could receive all users and organizations that needs validating but the problem is that I do validation by checking Dev.to API so if I would try to do all checks in a single API request it could timeout - this is why one-by-one value is being validated here.

Reply:
What I meant was that by using `IF` and `ELSE IF`, the organization logic/code will only be executed if there is no `req.query.username`, and I wasn't sure if that's intended

Reply:
Yes this was intented - if `req.query.username` is passed we check if this is a valid Dev.to user - if `req.query.organization` is passed we check if this is a valid Dev.to organization.